site/src/components/Resources/AppLink/AppLink.tsx

```diff
@@ -98,11 +98,19 @@ export const AppLink: FC<AppLinkProps> = ({ app, workspace, agent }) => {
             canClick
               ? (event) => {
                   event.preventDefault();
-                  window.open(
-                    href,
-                    Language.appTitle(appDisplayName, generateRandomString(12)),
-                    "width=900,height=600",
-                  );
+                  // This is an external URI like "vscode://", so
+                  // it needs to be opened with the browser protocol handler.
+                  if (app.external && !app.url.startsWith("http")) {
+                    // If the protocol is external the browser does not
+                    // redirect the user from the page.
+                    window.location.href = href
```

Comment:
I'd say we should just not prevent default and let the browser do it's thing

Reply:
also I feel like this should be an `||` not a `&&` right? if it was labeled as external by the template, _or_ it's not an http/https link, then we cannot open it as a new window

Reply:
The browser does the redirect but leaves an empty window artifact.

Reply:
An app can be external and not HTTP(s), in which case we'd want to open it in a new window I believe.

Reply:
Yes that's the current behavior. We are just adding an exception for non http ones. 
Having a window or tab is another debate. But let's leave that for now.

Reply:
> The browser does the redirect but leaves an empty window artifact.

oh because of the `_blank`, right? we could make that value depend on `app.external` to fix that.

Reply:
> An app can be external and not HTTP(s), in which case we'd want to open it in a new window I believe.

but the new window will still be a browser, only capable of opening http and https links. I'd expect that anything that's not http should not be opened in a new window

Reply:
I think if it's not labeled as external then we'd technically want to proxy it through the workspace, which I agree is weird, but don't wanna have a shot at breaking anyone right now before launch.